proof/dleq.go

```diff
@@ -0,0 +1,163 @@
+package proof
+
+import (
+	"errors"
+
+	"github.com/dedis/crypto/abstract"
+	"github.com/dedis/crypto/hash"
+	"github.com/dedis/crypto/random"
+)
+
+// This package provides functionality to create and verify non-interactive
+// zero-knowledge (NIZK) proofs for the equality (EQ) of discrete logarithms (DL).
+
+// DLEQ resembles a NIZK dlog-equality proof. Allows to handle multiple proofs.
+type DLEQ struct {
+	suite abstract.Suite
+	Base  []DLEQBase
+	Core  []DLEQCore
+}
+
+// DLEQBase contains the base points against which the core proof is created/verified.
+type DLEQBase struct {
```

Comment:
I didn't verify, so I just ask (we need an acronym for that ;): why are the fields of `DLEQBase` private, while the fields of `DLEQCore` are public?